src/wallet/wallet.h

```diff
@@ -146,6 +151,94 @@ struct COutputEntry
     int vout;
 };
 
+/** An note outpoint */
+class JSOutPoint
+{
+public:
+    // Transaction hash
+    uint256 hash;
+    // Index into CTransaction.vjoinsplit
+    size_t js;
+    // Index into JSDescription fields of length ZC_NUM_JS_OUTPUTS
+    uint8_t n;
```

Comment:
I think "unsigned char" is used in ``NoteDecryption`` etc.; it should be consistent (perhaps a typedef). Does not block this PR.